comp/logs/agent/config/endpoints.go

```diff
@@ -56,9 +58,142 @@ type Endpoint struct {
+		RecoveryReset:           logsConfig.senderRecoveryReset(),
+		useSSL:                  logsConfig.logsNoSSL(),
+		isReliable:              true, // by default endpoints are reliable
+	}
+}
+
+// The setting from 'logs_config.additional_endpoints' is directly unmarshalled from the configuration into a
+// []unmarshalEndpoint and do not use the constructors. In this case, apiKeyGetter is initialized to returned the API
+// key from the loaded data instead of 'api_key'/'logs_config.api_key'.
+
+func loadTCPAdditionalEndpoints(main Endpoint, l *LogsConfigKeys) []Endpoint {
+	additionals := l.getAdditionalEndpoints()
+
+	newEndpoints := make([]Endpoint, 0, len(additionals))
+	for _, e := range additionals {
+		newE := NewEndpoint(e.APIKey, e.Host, e.Port, false)
+
+		newE.UseCompression = e.UseCompression
+		newE.CompressionLevel = e.CompressionLevel
+		newE.ProxyAddress = l.socks5ProxyAddress()
+		newE.isReliable = e.IsReliable == nil || *e.IsReliable
+		newE.ConnectionResetInterval = e.ConnectionResetInterval
+		newE.BackoffFactor = e.BackoffFactor
+		newE.BackoffBase = e.BackoffBase
+		newE.BackoffMax = e.BackoffMax
+		newE.RecoveryInterval = e.RecoveryInterval
+		newE.RecoveryReset = e.RecoveryReset
+		newE.Version = e.Version
+		newE.TrackType = e.TrackType
+		newE.Protocol = e.Protocol
+		newE.Origin = e.Origin
+
+		if e.UseSSL == nil {
+			newE.useSSL = main.useSSL
+		} else {
+			newE.useSSL = *e.UseSSL
+		}
+		newEndpoints = append(newEndpoints, newE)
+	}
+	return newEndpoints
+}
+
+func loadHTTPAdditionalEndpoints(main Endpoint, l *LogsConfigKeys, intakeTrackType IntakeTrackType, intakeProtocol IntakeProtocol, intakeOrigin IntakeOrigin) []Endpoint {
+	additionals := l.getAdditionalEndpoints()
+
+	newEndpoints := make([]Endpoint, 0, len(additionals))
+	for _, e := range additionals {
+		newE := NewEndpoint(e.APIKey, e.Host, e.Port, false)
+
+		newE.UseCompression = main.UseCompression
+		newE.CompressionLevel = main.CompressionLevel
+		newE.ProxyAddress = e.ProxyAddress
+		newE.isReliable = e.IsReliable == nil || *e.IsReliable
+		newE.ConnectionResetInterval = e.ConnectionResetInterval
+		newE.BackoffFactor = main.BackoffFactor
+		newE.BackoffBase = main.BackoffBase
+		newE.BackoffMax = main.BackoffMax
+		newE.RecoveryInterval = main.RecoveryInterval
+		newE.RecoveryReset = main.RecoveryReset
+		newE.Version = e.Version
+		newE.TrackType = e.TrackType
+		newE.Protocol = e.Protocol
+		newE.Origin = e.Origin
+
+		if e.UseSSL == nil {
+			newE.useSSL = main.useSSL
+		} else {
+			newE.useSSL = *e.UseSSL
+		}
+
+		if newE.Version == 0 {
+			newE.Version = main.Version
+		}
+		if newE.Version == EPIntakeVersion2 {
+			newE.TrackType = intakeTrackType
+			newE.Protocol = intakeProtocol
+			newE.Origin = intakeOrigin
+		}
+
+		newEndpoints = append(newEndpoints, newE)
+	}
+	return newEndpoints
+}
+
+// GetAPIKey returns the API for the Endpoint.
+func (e *Endpoint) GetAPIKey() string {
```

Comment:
The convention that this PR introduces is that `Get` is in the method name because it will start returning new values once they get updated dynamically. I would spell this out in the comment:

```
// GetAPIKey returns the latest API Key for the Endpoint, including when it gets updated
```